### ISSUE TITLE
Number spinners become misaligned with ?stringTest=dynamic

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/fourier-making-waves#250]
**Test device**
MacBook Air M1 chip

**Operating System**
15.1.1

**Browser**
Safari 18.1.1

**Problem description**
For https://github.com/phetsims/qa/issues/1178, on the Discrete screen, the Wavelength and Period number spinners can become misaligned with ?stringTest=dynamic. This doesn't happen when I cycle through all of the locales on main.

**Steps to reproduce**
1. Add ?stringTest=dynamic to the end of the url
2. Select the Discrete Screen
3. Press the down arrow a bunch of times. 

**Visuals**
<img width="310" alt="Screenshot 2024-12-02 at 3 32 56 PM" src="https://github.com/user-attachments/assets/e0d3c21e-98f2-4c1d-9bb5-dbc39e50a6c3">

**Comment by Nancy-Salpepi** [comment]
Also, should there be a minimum height for the number displays? They can get oddly small. 
<img width="261" alt="Screenshot 2024-12-02 at 3 50 05 PM" src="https://github.com/user-attachments/assets/f060f2f1-1b69-4794-8b7c-9c7201e5dd9f">

**Comment by pixelzoom** [comment]
Reproduced in main.

**Comment by arouinfar** [comment]
> Also, should there be a minimum height for the number displays? They can get oddly small.

I think that's just a natural consequence of dynamic layout -- the height of the NumberDisplay depends on the height of the text. I don't think a reasonable translator would use more than 1-2 characters, so this situation is unlikely to happen in practice.

**Comment by pixelzoom** [comment]
> Also, should there be a minimum height for the number displays? They can get oddly small.

What @arouinfar said.  And in this case, we're not directly creating the NumberDisplays.  We are creating NumberControls, the NumberDisplays are subcomponets of the NumberControls.

That said, I see what I can do.

**Comment by pixelzoom** [comment]
> ...  We are creating NumberControls, ...

Correction... These are not NumberControls.  They are NumberSpinners, configured with `arrowsPosition: 'leftRight'`.

**Comment by pixelzoom** [comment]
The problem here is that support for dynamic layout has not been added to NumberSpinner.  So even though I'm using an AlignGroup to make all spinners have the same effective size, their NumberDisplays are not stretching to fill the space.  I've opened https://github.com/phetsims/sun/issues/918 to address this, but I've indicated that it's not blocking.  In the meantime, I'll leave this issue open.  But we can certainly proceed with RC testing and even publication if it's not addressed.
